tfe_hardware/qcodes_instrument_drivers/Sinko/AD5760.py

```diff
@@ -0,0 +1,170 @@
+'''
+A simple driver to control the AD5760 current source
+'''
+
+__author__ = "Kaushik Singirikonda"
+__email__ = 'ks105@illinois.edu'
+
+from functools import partial
+from typing import Optional, Union, Any
+import time,sys
+import clr  # install pythonnet
+
+from qcodes.instrument.parameter import DelegateParameter
+from qcodes import (Instrument, validators as vals)
+from qcodes.instrument.channel import InstrumentChannel
+from qcodes.utils.validators import Numbers, Bool, Enum, Ints
+
+'''
+To use this current source, you first have to install and setup the ACE software. Upon connecting to the AD5760
+board with your computer using the software, you have to initialize a setup on python to talk to the software to
+enable writing.
+
+Following this, you can simply run this code to operate a stable current source.
+
+
+'''
+
+
+class AD5760(Instrument):
+
+    def __init__(self, name: str, host_name: str, path: str, syspath: str, terminator: str = "\n",
+                 **kwargs: Any) -> None:
+        super().__init__(name, **kwargs)
+        self.connect_board(host_name, path, syspath)
+
+        self.add_parameter('output',
+                           label='Output State',
+                           get_cmd=self.state,
+                           set_cmd=lambda x: self.on() if x else self.off(),
+                           val_mapping={
+                               'off': 0,
+                               'on': 1,
+                           })
+        
+        self.add_parameter('voltage',
+                           label='Voltage',
+                           unit='V',
+                           set_cmd=self._get_set_volt,
+                           get_cmd=self._get_volt,
+                           vals= Numbers(min_value=-10, max_value=10)
+                           )
+        
+        self.add_parameter('current',
+                           label='Current',
+                           unit='A',
+                           set_cmd=self._get_set_current,
+                           get_cmd=self._get_current,
+                           vals= Numbers(min_value=-0.1, max_value=0.1))
+        
+    def on(self) -> None:
+        self.write('OUTPUT 1')
+        self.measure._output = True
+        
+    def off(self) -> None:
+        self.write('OUTPUT 0')
+        self.measure._output = False
+
+    def state(self):
+        state = self.ask()
+        if state is not None:
+            return bool(1)
+        else:
+            return bool(0)
```

Comment:
self.ask() will never return `None`, because you cannot convert `None` to `int` or `float`, if asking for a number ever returns `None` it needs to be properly handled in the ask function.

`bool(0)` or `bool(1)` work on python but utilizing the native `True` or `False` makes it easier to read.